autogen/agentchat/conversable_agent.py

```diff
@@ -1175,17 +1366,22 @@ def _format_json_str(jstr):
     def execute_function(self, func_call, verbose: bool = False) -> Tuple[bool, Dict[str, str]]:
         """Execute a function call and return the result.
 
-        Override this function to modify the way to execute a function call.
+        Override this function to modify the way to execute function and tool calls.
 
         Args:
-            func_call: a dictionary extracted from openai message at key "function_call" with keys "name" and "arguments".
+            func_call: a dictionary extracted from openai message at "function_call" or "tool_calls" with keys "name" and "arguments".
 
         Returns:
             A tuple of (is_exec_success, result_dict).
             is_exec_success (boolean): whether the execution is successful.
             result_dict: a dictionary with keys "name", "role", and "content". Value of "role" is "function".
+
+        "function_call" deprecated as of [OpenAI API v1.1.0](https://github.com/openai/openai-python/releases/tag/v1.1.0)
+        See https://platform.openai.com/docs/api-reference/chat/create#chat-create-function_call
         """
         func_name = func_call.get("name", "")
+        # OpenAI API blows up if name doesn't match /^[a-zA-Z0-9_-]{1,64}$/
+        func_name = self._normalize_name(func_name)
```

Comment:
Should validate the function name when the function is added to the agent.

Reply:
yeah turns out these are overkill, everything is covered by _append_oai_message